torch/csrc/autograd/function.h

```diff
@@ -56,7 +56,7 @@ struct MakeNextFunctionList : IterArgs<MakeNextFunctionList> {
 inline bool any_variable_requires_grad(const variable_list& variables) {
   return std::any_of(
       variables.begin(), variables.end(), [](const Variable& variable) {
-        return variable.requires_grad();
+        return variable.defined() and variable.requires_grad();
```

Comment:
Did you mean to use `and` instead of `&&`?

Reply:
`and` is equivalent to `&&` I believe

Reply:
Yeah but we don't use it anywhere else, and [no one uses it in practice](https://stackoverflow.com/questions/14740864/any-downside-to-using-the-and-operator-vs-the-operator)

Reply:
We use it pretty much everywhere ;)